_layouts/focus-area.html

```diff
@@ -62,89 +62,30 @@ <h2 class="alt-h2 text-center mb-3 mt-lg-6" id="projects">{{ page.short_title |
 {% endif %}
 
 
-{% assign sorted_presentations = site.data['sorted_presentations'] %}
 {% assign page_focus_area = page.name | basename %}
-
-{% assign pres_list = "" | split: "," %}
-{% for pres in sorted_presentations %}
-  {% if pres.focus-area contains page_focus_area %}
-    {% assign pres_list = pres_list | push: pres %}
-  {% endif %}
-{% endfor %}
+{% assign pres_list = site.data['sorted_presentations'] | where_exp: "item", "item.focus-area contains page_focus_area"  %}
 
 {% if pres_list.size > 0 %}
 <section class="">
   <div class="container-lg p-responsive py-5 py-md-6 ">
     <h2 class="alt-h2 text-center mb-3 mt-lg-6" id="pres">{{page.short_title | upcase }} Presentations</h2>
-
-<ul>
-{% assign max_disp = 15 %}
-{% assign counter = 0 %}
-{% for talk in pres_list %}
-  {% assign counter = counter | plus: 1 %}
-  <li>
-    {%- include print_pres.html talk=talk -%}
-  </li>
-  {% if counter == max_disp %}
-    </ul>
-    <p>[expand]</p>
-    <ul>
-  {% endif %}
-{% endfor %}
-</ul>
-{% if counter >= max_disp %}
-<p>[/expand]</p>
-{% endif %}
+    {% expandable pres_list 15 %}
+      {% include print_pres.html talk=expandable %}
+    {% endexpandable %}
 </div>
 </section>
 {% endif %}
 
-{% assign sorted_publications = site.data['sorted_publications'] %}
-
-{% assign pubs = "" | split: "," %}
-{% for pub in sorted_publications %}
-  {% assign page_st = page.name | basename %}
-  {% if pub.focus-area contains page_st %}
-    {% assign pubs = pubs | push: pub %}
-  {% endif %}
-{% endfor %}
+{% assign pubs = site.data['sorted_publications'] | where_exp: "item", "item.focus-area contains page_focus_area" %}
 
 {% if pubs.size > 0 %}
-
-
 <div class="container-lg p-responsive py-5 py-md-6 ">
     <h2 class="alt-h2 text-center mb-3 mt-lg-6" id="pubs">{{ page.short_title | upcase }} Publications</h2>
-
-
-<ul>
-{% assign max_disp = 15 %}
-{% assign counter = 0 %}
-  {% for pub in pubs %}
-    {% assign counter = counter | plus: 1 %}
-    {% if pub.citation-count and pub.citation-count > 0 %}
-      {% assign cited = " [" | append: pub.citation-count | append: " citations]" %}
-      {% if pub.inspire-id %}
-        {% capture cited %} <a href='http://inspirehep.net/record/{{pub.inspire-id}}/citations'>{{cited}}</a>{% endcapture %}
-      {% endif %}
-    {% else %}
-      {% assign cited = "" %}
-    {% endif %}
-    <li> {% include print_pub.html pub=pub %} </li>
-    {% if counter == max_disp %}
-    </ul>
-    <p>[expand]</p>
-    <ul>
-    {% endif %}
-  {% endfor %}
-</ul>
```

Comment:
@davidlange6 I've moved this into a new Ruby block "expandable".

Reply:
the code certainly looks better:)